src/App.tsx

```diff
@@ -1,10 +1,54 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Sort } from './utils/Sort';
+import { TodosList } from './components/TodosList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
 
-const USER_ID = 0;
+const USER_ID = 10360;
 
 export const App: React.FC = () => {
+  const [newTodo, setNewTodo] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortType, setSortType] = useState<Sort>(Sort.All);
+  const [hasError, setHasError] = useState(false);
+
+  const loadTodos = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setHasError(true);
+      setTimeout(() => {
+        setHasError(false);
+      }, 3000);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const sortingTodos = (type: Sort) => {
+    switch (type) {
+      case Sort.Active:
+        return todos.filter(todo => !todo.completed);
+      case Sort.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const isAnyActiveTodo = todos.some(todo => !todo.completed);
+  const isAnyCompletedTodo = todos.some(todo => todo.completed);
+  const itemsLeftToComplete = todos.filter(todo => !todo.completed).length;
+  const sortedTodos = sortingTodos(sortType);
```

Comment:
let's use `useMemo` hook

Reply:
Unfortunately, I can't apply the useMemo to sortedTodos, because in this case, the todo are not displayed at all on initial load.